src/server/_query.py

```diff
@@ -375,7 +410,19 @@ def __str__(self):
         group_by = f"GROUP BY {_join_l(self.group_by)}" if self.group_by else ""
         index = f"USE INDEX ({self.index})" if self.index else ""
 
-        return f"SELECT {self.fields_clause} FROM {self.table} {index} {self.subquery} {where} {group_by} {order}"
+        # if no signal array, assemble the sql and return
+        if not self.signal_array:
+            command = f"SELECT {self.fields_clause} FROM {self.table} {index} {self.subquery} {where} {group_by}" 
+        else:
+            # if there is a signal array, concatenate signals with UNION ALL
+            command = f"SELECT {self.fields_clause} FROM {self.table} {index} {self.subquery} {where} AND {self.signal_array[0]} {group_by}"
+            i = 1
+            while i < len(self.signal_array):
+                command += f"\nUNION ALL\nSELECT {self.fields_clause} FROM {self.table} {index} {self.subquery} {where} AND {self.signal_array[i]} {group_by}"
```

Comment:
Are the linebreaks for readability while debugging? if so, they should be removed before we merge, since they'll probably mess up the logs

Reply:
how does `GROUP BY` interact with `UNION ALL`? if we're grouping by something other than source/signal, will the collected results be grouped correctly or does `UNION` impose an implicit grouping by the `signal_array` column(s)?

Reply:
Linebreaks removed. Group by and Order by added at the end of the query.